### PR TITLE
WDP240901-14

### PR DESCRIPTION
**Pull request by patrykbn**
Added RWD for tablets and mobile devices.
Added dropdown menu for mobile devices

https://projects.kodilla.com/secure/RapidBoard.jspa?rapidView=239&projectKey=WDP240901&view=detail&selectedIssue=WDP240901-14

### CODE REVIEW

File: src/components/layout/MenuBar/MenuBar.module.scss [hunk 1]
```diff
@@ -48,4 +48,92 @@
...
+      }
+    }
+  }
+
+  .dropdown {
+    position: absolute;
+    top: 60px;
+    left: 0;
+    right: 0;
+    background-color: #ffffff;
```
Comment (matwjablonski) [inline src/components/layout/MenuBar/MenuBar.module.scss:95]: color from variables

File: src/components/layout/MenuBar/MenuBar.module.scss [hunk 2]
```diff
@@ -48,4 +48,92 @@
...
+    box-shadow: 0 2px 10px rgba(0, 0, 0, 0.1);
```
Comment (matwjablonski) [inline src/components/layout/MenuBar/MenuBar.module.scss:96]: color from variables

**Review by matwjablonski — CHANGES_REQUESTED** [review]

File: src/components/layout/MenuBar/MenuBar.module.scss (hunk 2 above)
Comment (patrykbn) [reply src/components/layout/MenuBar/MenuBar.module.scss]: color variables added

File: src/components/layout/MenuBar/MenuBar.module.scss (hunk 1 above)
Comment (patrykbn) [reply src/components/layout/MenuBar/MenuBar.module.scss]: color variables added

**Comment by dave8git** [comment]
I think task mentioned that on small screens dropdown menu should hide, and only after clicking on an icon, dropdown menu should appear and cover part of the page: 
Screenshot 2024-09-24 092730
Please hide dropdown menu for smaller screen sizes.

**Comment by dave8git** [comment]
> > I think task mentioned that on small screens dropdown menu should hide, and only after clicking on an icon, dropdown menu should appear and cover part of the page: Screenshot 2024-09-24 092730 Please hide dropdown menu for smaller screen sizes.
> 
> The task was for the search bar to be under the Menu buttons for tablets, and for mobile devices the menu to change to a clickable dropdown menu on the right of the searchbar. I don't see a reason to hide the dropdown menu for mobile devices. How would users access the menu options?

What would be the reason to have two category dropdowns then? It's redundant. Why would you have one in select category and another one in dropdown menu that shows when click the icon? It should be visible on small screens, that's why you shouldn't have large box just icon and then menu that hides part of the screen when it's open.

**Comment by patrykbn** [comment]
> > > I think task mentioned that on small screens dropdown menu should hide, and only after clicking on an icon, dropdown menu should appear and cover part of the page: Screenshot 2024-09-24 092730 Please hide dropdown menu for smaller screen sizes.
> > 
> > 
> > The task was for the search bar to be under the Menu buttons for tablets, and for mobile devices the menu to change to a clickable dropdown menu on the right of the searchbar. I don't see a reason to hide the dropdown menu for mobile devices. How would users access the menu options?
> 
> What would be the reason to have two category dropdowns then? It's redundant. Why would you have one in select category and another one in dropdown menu that shows when the user clicks on the icon? It should be visible on small screens, that's why you shouldn't have large box just icon and then menu that hides part of the screen when it's open. You should hide large category dropdown menu, and show only the icons, if someone wants to see categories they will click on the icon and menu will come down.
> 
> "Na tabletach wyszukiwanie ma być pod menu, a na mniejszych ekranach ma być wyszukiwanie **i obok niego ikona** mobilnego menu, które po rozwinięciu ma pokazywać dropdown przykrywający treść strony."

Dawidzie, napiszę Ci po polsku, żeby nie było żadnych wątpliwości,
Zauważ, że select a category i dropdown menu to dwa kompletnie inne elementy.
Do dropdown menu "chowają się" przyciski menu które są widoczne dla większych ekranów. (tj. HOME FURNITURE CHAIR TABLE SOFA BEDROOM BLOG)
Select a category z kolei dotyczy kategorii w której chcemy wyszukać produkt- nie podstrony z MENU. Tutaj po rozwinięciu select a category otrzymasz kategorie BED CHAIR SOFA TABLE DINING. Teraz tego nie widać, ale w zadaniu, które sam z resztą robiłeś kategorie są widoczne po najechaniu na element.

Prosiłbym Cie na przyszłość, żeby czytać zadania dokładnie i przemyślenie treści, bo oboje tracimy na takie wyjaśnianie cenny czas.

**Comment by dave8git** [comment]
> > > > I think task mentioned that on small screens dropdown menu should hide, and only after clicking on an icon, dropdown menu should appear and cover part of the page: Screenshot 2024-09-24 092730 Please hide dropdown menu for smaller screen sizes.
> > > 
> > > 
> > > The task was for the search bar to be under the Menu buttons for tablets, and for mobile devices the menu to change to a clickable dropdown menu on the right of the searchbar. I don't see a reason to hide the dropdown menu for mobile devices. How would users access the menu options?
> > 
> > 
> > What would be the reason to have two category dropdowns then? It's redundant. Why would you have one in select category and another one in dropdown menu that shows when the user clicks on the icon? It should be visible on small screens, that's why you shouldn't have large box just icon and then menu that hides part of the screen when it's open. You should hide large category dropdown menu, and show only the icons, if someone wants to see categories they will click on the icon and menu will come down.
> > "Na tabletach wyszukiwanie ma być pod menu, a na mniejszych ekranach ma być wyszukiwanie **i obok niego ikona** mobilnego menu, które po rozwinięciu ma pokazywać dropdown przykrywający treść strony."
> 
> Dawidzie, napiszę Ci po polsku, żeby nie było żadnych wątpliwości, Zauważ, że select a category i dropdown menu to dwa kompletnie inne elementy. Do dropdown menu "chowają się" przyciski menu które są widoczne dla większych ekranów. (tj. HOME FURNITURE CHAIR TABLE SOFA BEDROOM BLOG) Select a category z kolei dotyczy kategorii w której chcemy wyszukać produkt- nie podstrony z MENU. Tutaj po rozwinięciu select a category otrzymasz kategorie BED CHAIR SOFA TABLE DINING. Teraz tego nie widać, ale w zadaniu, które sam z resztą robiłeś kategorie są widoczne po najechaniu na element.
> 
> Prosiłbym Cie na przyszłość, żeby czytać zadania dokładnie i przemyślenie treści, bo oboje tracimy na takie wyjaśnianie cenny czas.

Patryku, masz rację. Przyjrzałem się treści i to co mówisz jest prawdą. Na pierwszy rzut oka patrzyłem na zachowanie RWD i nie pasuje mi to, że przez dwa okna wyszukiwania (select-dropdown, i input-search) na mobilkach rozjeżdża się widok. Popatrz sam jak to wygląda na Iphone XR: 

image

**Review by dave8git — APPROVED** [review]

**Comment by dave8git** [comment]
> @dave8git I edited the styles to look good on viewports of a minimum of about 410px width. Let me know if you think it looks good now. Thank you!

I wonder if this line doesn't suggest that search box should be under the menu. On year design, they are side by side. I think one should be below?  Do you agree? I see two steps here. 1. On tablets searchbox should be below menu, and 2. on mobile, there should be only search box and  icon which will open dropdown covering portion of the page. "Na tabletach *1.* wyszukiwanie ma być pod menu, "2.* a na mniejszych ekranach ma być wyszukiwanie i obok niego ikona mobilnego menu, które po rozwinięciu ma pokazywać dropdown przykrywający treść strony."

**Review by dave8git — APPROVED** [review]

**Comment by matwjablonski** [comment]
conflicts

**Review by dave8git — APPROVED** [review]

**Review by matwjablonski — APPROVED** [review]

**Pull request merged**